### ISSUE TITLE
Navigation does not start correct on any point of a track

### ISSUE THREAD

**Issue by C6competition** [osmandapp/OsmAnd#18154]
The setting for navigation on tracks in this issue is "NO RECALCULATION" and "POINT OF THE TRACK TO NAVIGATE -> NEAREST POINT".
The track , attached to the road and calculated, is this:

[Bodenw Löwenherz Blomberg Neu.gpx.txt](https://github.com/osmandapp/OsmAnd/files/12606779/Bodenw.Lowenherz.Blomberg.Neu.gpx.txt)

Asuming the actual position is somewhere on the track and you want to start navigation on the track from this position, these two different results will happen, depending on the distance, you are away from the startpoint of the track.

1.
On this screenshot you see everything is happening like expected. the track before the actual position is shown in blue (like it's already done), while the track in front is shown in red ( the part which has to be followed). As the track is 311 km long, the values in the panels and the announcement seemed to be correct as well.

![Start 1](https://github.com/osmandapp/OsmAnd/assets/99248549/8c69bbf2-cb88-4478-8dd5-09ce5672e83d)

2.
Changing the actual position only about 1 km further down on of the same street,
gives a complete different result. Now the whole track is shown in red colour (this is normally the part you have to follow). The announcement tells you, you are away from the track since 50 km! In the right panel the distance of the track is extended by 51 km to 362 km. It looks like the navigation started at the beginning of the track, not at the current position.

![Start 2](https://github.com/osmandapp/OsmAnd/assets/99248549/9c2dcead-0016-4c29-a473-87b6630a45e3)

Up to the end of the track, the same happens again: if your position is close enough to the finish (the distance is again about 50 km), you get what you expect.
A few hundred meters before, the same mistake happens, as at the beginning of the track. See the added 2 screenshots.

![End 1](https://github.com/osmandapp/OsmAnd/assets/99248549/aaf6eba6-d416-4619-b3ab-aece95457793)

![End 2](https://github.com/osmandapp/OsmAnd/assets/99248549/01d7ec38-1325-4591-a8b7-84694cc7bea8)

Don't mix this issue up with the one, with a similar title #14796
There was a different reason for it and it's solved!

Version 4.5.10
Android 11.0
Blackview  9900E

**Comment by C6competition** [comment]
Is anybody taking care about this issue?

**Comment by DmitryAlexei** [comment]
OsmAnd~ 4.6.0#871m, released: 2023-10-13
Cannot reproduce this issue using provided track and following settings:
- in route parameters (navigation > Navigation settings > Route parameters) options *Minimal distance to recalculate route* and *In case of reverse direction* are disabled (see screenshot)
- provided track is attached to the roads in navigation mode
- location is simulated using provided gpx and created routes (tried different variants)

In case of deviation from the track when recalculation options are enabled, the route is sometimes calculated to the nearest point in the begging of the track (see video), that change the length and ETA. 
<table style="width:100%">
  <tr>
    <th>Settings</th>
    <th>Actual result</th>
    <th>When recalculation enabled</th>
  </tr>
  <tr>
    <td><img src="https://github.com/osmandapp/OsmAnd/assets/116079541/04fa0e71-905e-41cf-b4ea-3810a6565839"  width="300" height="auto"/> </td>
    <td><img src="https://github.com/osmandapp/OsmAnd/assets/116079541/c23e4b9d-107b-4e78-9570-e90d648a4bdc"  width="300" height="auto"/> </td>
    <td><video src="https://github.com/osmandapp/OsmAnd/assets/116079541/66df3abf-06f2-463c-a1c9-214f8b711fa8"  width="300" height="auto"/> </td>
  </tr>
</table>

**Comment by C6competition** [comment]
After investigating the issue again, I found the difference between the 2 situations.

In one case ( farer away from the Start/End of the track) the system insertes in the start navigation window the STARTPOINT OF THE TRACK as the startpoint for navigation!

In the other case (closer to the Start/End of the track) the sytem inserts in the start navigation window the ACTUAL POSITION as the startpoint for navigation!



As mostly you start navigation at the beginning of a track (the actual position is inserted) this stays in my mind and I didn't pay attention to the start point any more. My fault!! Sorry for the work I caused you.

But for the future, I suggest, as default always use the ACTUAL POSITION at the start position, because mostely You start navigation from the point, where you are at the moment.
